### PR TITLE
HDDS-9049. Use OM service ID from Ozone address for recursive delete

### PR DESCRIPTION
**Pull request by ashishkumar50**
## What changes were proposed in this pull request?

OM ServiceId can be fetched from address and can be used to form Ozone address for fs URI. This can avoid giving service ID explicitly in option while running recursive delete volume command. Already this has been done for delete bucket command handler. Refactored getting serviceId from common place. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9049

## How was this patch tested?

Update existing integration test and verify.

### CODE REVIEW

**Comment by ashishkumar50** [comment]
@adoroszlai @sadanand48 , Please help to review.

**Review by sadanand48 — COMMENTED** [review]
Thanks @ashishkumar50 for the patch . LGTM

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/volume/DeleteVolumeHandler.java [hunk 1]
```diff
@@ -92,16 +85,17 @@ protected void execute(OzoneClient client, OzoneAddress address)
...
+          Collection<String> serviceIds = getConf().getTrimmedStringCollection(
+              OZONE_OM_SERVICE_IDS_KEY);
+          if (serviceIds.size() == 1) {
             // Only one OM service ID configured, we can use that
+            // If more than 1, it will fail in createClient step itself
             omServiceId = serviceIds.iterator().next();
+          } else {
+            omServiceId = getConf().get(OZONE_OM_ADDRESS_KEY);
           }
         }
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/volume/DeleteVolumeHandler.java:100]: This is same logic as in `DeleteBucketHandler`.  Can you please extract and reuse?

Comment (ashishkumar50) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/volume/DeleteVolumeHandler.java]: Thanks, extracted in common place and reused.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @ashishkumar50 for the patch, @sadanand48 for the review.

**Pull request merged**